### ISSUE TITLE
Q: Supressed NullPointerException

### ISSUE THREAD

**Issue by patric-r** [brettwooldridge/HikariCP#673]
During a database downtime, an application which uses HikariCP throws the exception stated below during pool startup.
Due to `checkFailFast`, it's clear that the pool cannot be initialized and that an exception will be thrown.

**Question:**
Is the `Suppressed: java.lang.NullPointerException` part expected in such a case or is this a bug?

```
com.zaxxer.hikari.pool.HikariPool$PoolInitializationException: Failed to initialize pool: IO Error: Socket read timed out
 	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:524) ~[HikariCP-2.4.5.jar:?]
 	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:118) ~[HikariCP-2.4.5.jar:?]
 	at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:71) ~[HikariCP-2.4.5.jar:?]
 [...] 
 Caused by: java.sql.SQLRecoverableException: IO Error: Socket read timed out
 	at oracle.jdbc.driver.T4CConnection.logon(T4CConnection.java:743) ~[oracle-jdbc7-12.1.0.2.jar:12.1.0.2.0]
 	at oracle.jdbc.driver.PhysicalConnection.connect(PhysicalConnection.java:666) ~[oracle-jdbc7-12.1.0.2.jar:12.1.0.2.0]
 	at oracle.jdbc.driver.T4CDriverExtension.getConnection(T4CDriverExtension.java:32) ~[oracle-jdbc7-12.1.0.2.jar:12.1.0.2.0]
 	at oracle.jdbc.driver.OracleDriver.connect(OracleDriver.java:566) ~[oracle-jdbc7-12.1.0.2.jar:12.1.0.2.0]
 	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:95) ~[HikariCP-2.4.5.jar:?]
 	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:101) ~[HikariCP-2.4.5.jar:?]
 	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:315) ~[HikariCP-2.4.5.jar:?]
 	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:514) ~[HikariCP-2.4.5.jar:?]
 	... 12 more
 	Suppressed: java.lang.NullPointerException
 		at com.zaxxer.hikari.pool.HikariPool.shutdown(HikariPool.java:219) ~[HikariCP-2.4.5.jar:?]
 		at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:518) ~[HikariCP-2.4.5.jar:?]
 		at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:118) ~[HikariCP-2.4.5.jar:?]
 		at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:71) ~[HikariCP-2.4.5.jar:?]
  [...]
```

**Comment by johnou** [comment]
The bug was fixed in 2.4.6.

**Comment by patric-r** [comment]
Thank you.
